### PR TITLE
[RECO] Fixes needed for new root cxxmodules

### PR DESCRIPTION
**Pull request by smuzaffar**
- Added missing dependency 
```
>grep -R DataFormats/SiStripDigi DataFormats/SiStripCluster
DataFormats/SiStripCluster/interface/SiStripCluster.h:#include "DataFormats/SiStripDigi/interface/SiStripDigi.h"
```
- Removed usage of pattern `ROOT::Math::*` from https://github.com/cms-sw/cmssw/blob/c52e2819d7a2bca52273e9c36fc99622313eff71/DataFormats/Math/src/classes_def.xml#L7

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: DataFormats/Math/src/classes.h [hunk 1]
```diff
@@ -3,6 +3,8 @@
 #define SMATRIX_USE_CONSTEXPR
 #include <Math/SVector.h>
 #include <Math/SMatrix.h>
+#include <Math/MinimTransformVariable.h>
+#include <TGraphDelaunay2D.h>
```
Comment (smuzaffar) [inline DataFormats/Math/src/classes.h]: @slava77 , these can be avoid if we replace the `ROOT::Math::*` pattern.

Comment (slava77) [reply DataFormats/Math/src/classes.h]: do you mean the patterns used in ```DataFormats/Math/src/classes_def.xml``` ?
There was some cleanup of dictionaries by the core team a while back to drop the wildcards.
Can that be revived/completed?

Comment (smuzaffar) [reply DataFormats/Math/src/classes.h]: yes the pattern in `DataFormats/Math/src/classes_def.xml` . I am testing the following right now.
```
-  <class pattern="std::vector<ROOT::Math::*>" />
+  <class pattern="std::vector<ROOT::Math::SMatrix<*>>" />
+  <class pattern="std::vector<ROOT::Math::MatRepStd<*>>" />
+  <class pattern="std::vector<ROOT::Math::RowOffsets<*>>" />
+  <class pattern="std::vector<ROOT::Math::SVector<*>>" />
```

**Comment by smuzaffar** [comment]
@slava77 , @perrotta I updated the PR and replace the usage of  pattern `ROOT::Math::*` in classes_def.xml

**Comment by slava77** [comment]
@cmsbuild please test

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/26883	c9711ac
- technical
- jenkins tests pass

**Comment by fabiocos** [comment]
+1

**Comment by slava77** [comment]
ah, so , in https://github.com/cms-sw/cmssw/pull/26883#issuecomment-494969047

> Additional comment: 'Compilation

was actually meant to sound a significant alarm. I did not notice it.
I'll try to pay more attention to the summaries.
Although perhaps a bit more verbose message would be better.

**Comment by smuzaffar** [comment]
It was meant to say "Compilation Warnings: Yes" which should have also added the "warning" label but due to bug in new pr build script it did not post the full message. I will fix this soon.

**Comment by fabiocos** [comment]
@smuzaffar @slava77 unfortunately I missed it as well...

**Pull request merged**